### PR TITLE
kubectl pg check

### PR DESCRIPTION
**Pull request by VineethReddy02**
This is the initial work on **postgres kubectl plugin**

### CODE REVIEW

File: kubectl-pg/LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
```
Comment (erthalion) [inline kubectl-pg/LICENSE]: Why the plugin comes with a different license than the operator itself?

Comment (VineethReddy02) [reply kubectl-pg/LICENSE]: I used cobra it has generated it. I need to do some cleanup didn't pay attention towards it. I will remove that.

File: kubectl-pg/main.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import "kubectl-pg/cmd"
```
[the remark this answers is not in the document]
Comment (VineethReddy02) [reply kubectl-pg/main.go]: Okay, I will change accordingly. I was developing plugin independently missed checking this.

File: kubectl-pg/cmd/delete.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// deleteCmd represents the delete command
+var deleteCmd = &cobra.Command{
+	Use:   "delete",
```
Comment (sdudoladov) [inline kubectl-pg/cmd/delete.go]: same here -> better descriptions

File: kubectl-pg/cmd/check.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
+// Copyright © 2019 NAME HERE <EMAIL ADDRESS>
```
Comment (sdudoladov) [inline kubectl-pg/cmd/check.go]: we will clarify the question with licences and keep you informed

**Comment by sdudoladov** [comment]
some general comments:
1. please give this PR a descriptive name, saying it is  a kubectl plugin and work-in-progress (WIP) ; see other PRs for examples of descriptive names
2. it should be possible to develop a plugin in a feature branch . why in your fork are you developing on the master branch ?

File: kubectl-pg/cmd/check.go (hunk 4 above)
Comment (VineethReddy02) [reply kubectl-pg/cmd/check.go]: Okay till then I will remove the license related content.

File: kubectl-pg/cmd/delete.go (hunk 3 above)
Comment (VineethReddy02) [reply kubectl-pg/cmd/delete.go]: I will add better comments for better readability.

File: kubectl-pg/cmd/create.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// createCmd represents the create command
+var createCmd = &cobra.Command{
+	Use:   "create",
```
[the remark this answers is not in the document]
Comment (VineethReddy02) [reply kubectl-pg/cmd/create.go]: I will follow kubectl level description on --help or -h

File: kubectl-pg/cmd/check.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"path/filepath"
+)
+
+const postgresCrdName string = "postgresqls.acid.zalan.do"
+
+// checkCmd represents the check command
+var checkCmd = &cobra.Command{
+	Use:   "check",
```
[the remark this answers is not in the document]
Comment (VineethReddy02) [reply kubectl-pg/cmd/check.go]: I will go for a better description.

File: kubectl-pg/cmd/check.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	"github.com/spf13/cobra"
+	apiextbeta1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"path/filepath"
+)
+
+const postgresCrdName string = "postgresqls.acid.zalan.do"
```
[the remark this answers is not in the document]
Comment (VineethReddy02) [reply kubectl-pg/cmd/check.go]: I will use the existing ones from the operator constants.

File: kubectl-pg/cmd/check.go [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"github.com/spf13/cobra"
+	apiextbeta1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//"k8s.io/client-go/kubernetes"
```
[the remark this answers is not in the document]
Comment (VineethReddy02) [reply kubectl-pg/cmd/check.go]: Okay.

File: kubectl-pg/cmd/check.go (hunk 4 above)
Comment (sdudoladov) [reply kubectl-pg/cmd/check.go]: @VineethReddy02 this is a legal document; nobody can remove them freely. If the library requires the resultant code to stay under the Apache license, it must stay under this licence

Comment (VineethReddy02) [reply kubectl-pg/cmd/check.go]: Okay @sdudoladov. I will not remove anything generated by cobra library regarding the license.

**Comment by VineethReddy02** [comment]
@sdudoladov I have created a new PR #579   with a feature branch and updated with all initial code review comments.

**Comment by FxKu** [comment]
closing this one in favor for #579

**Pull request closed without merging**